src/V3Ast.cpp

```diff
@@ -1082,26 +1082,97 @@ void AstNode::dumpPtrs(std::ostream& os) const {
 
 void AstNode::dumpTree(std::ostream& os, const string& indent, int maxDepth) const {
     static int s_debugFileline = v3Global.opt.debugSrcLevel("fileline");  // --debugi-fileline 9
-    os << indent << " " << this << '\n';
-    if (debug() > 8) {
-        os << indent << "     ";
-        dumpPtrs(os);
-    }
-    if (s_debugFileline >= 9) os << fileline()->warnContextSecondary();
-    if (maxDepth == 1) {
-        if (op1p() || op2p() || op3p() || op4p()) os << indent << "1: ...(maxDepth)\n";
-    } else {
+    if (v3Global.opt.dumpTreeDot()) {
+        os << this;
+        if (op1p()) {
+            if (op1p()->typeName() != "BASICDTYPE") {
+                os << "\tn" << cvtToHex(this) << " -> n" << cvtToHex(op1p()) << " ["
+                   << "fontsize=8 label=\"" << "op1" << "\""
+                   << " weight=" << 1 << " color=" << "red" << "];\n";
+            }
+        }
```

Comment:
```suggestion
        if (m_op1p) {
            os << "\tn" << cvtToHex(this) << " -> n" << cvtToHex(m_op1p) << " ["
               << "fontsize=8 label=\"" << "op1" << "\""
               << " weight=" << 1 << " color=" << "red" << "];\n";
        }
```

I don't think we want to omit anything from this dump, also node that `op1p()->typeName() != "BASICDTYPE"` is pointer comparison which might or might not do what you expect based on your toolchain.